blocks/subscriptions-block/features/checkout/_children/PaymentInfo/index.jsx

```diff
@@ -1,51 +1,175 @@
+	const [payment, setPayment] = useState({});
+
+	// load stripe key via payment details stripe key string
+	const { parameter2: stripeKey, parameter1: clientSecret } = payment;
 
 	const phrases = usePhrases();
 
 	const formErrorText = phrases.t("subscriptions-block.payment-error");
 	const formLabel = phrases.t("subscriptions-block.credit-card-information");
-	const formTitle = phrases.t("subscriptions-block.payment-information");
 	const submitText = phrases.t("subscriptions-block.submit-payment");
+	const payWithCardDividerLabel = phrases.t("subscriptions-block.payWithCard-label");
+
+	useEffect(() => {
+		if (stripeIntents?.paymentMethodID && !errorPaymentOption && !isInitialized) {
+				Sales.getCart().then((cart) => {
+					if (!cart?.items?.length) {
+						window.location.href = offerURL;
+						return;
+					} else {
+						const { country, email } = userInfo;
+						Sales.createNewOrder({ country }, email)
+							.then((order) => {
+								setOrderNumber(order.orderNumber);
+								Sales.initializePayment(order.orderNumber, stripeIntents?.paymentMethodID).then(
+									(paymentObject) => {
+										setPayment(paymentObject);
+									},
+								);
+							})
+							.catch((e) => console.error(e));
+					}
+				});
+			}
+	}, [stripeIntents, errorPaymentOption]);
 
-	// load stripe key via payment details stripe key string
 	useEffect(() => {
-		// stripe docs https://stripe.com/docs/stripe-js/react#elements-provider
-		loadStripe(stripeKey).then((newStripePromise) => setStripeInstance(newStripePromise));
-	}, [stripeKey]);
-
-	if (stripeInstance) {
-		// elements wrapper has to contain any stripe hooks
-		return (
-			<Elements stripe={stripeInstance}>
-				<PaymentForm
-					clientSecret={clientSecret}
-					formErrorText={formErrorText}
-					formLabel={formLabel}
-					formTitle={formTitle}
-					orderNumber={orderNumber}
-					paymentMethodID={paymentMethodID}
-					stripeInstance={stripeInstance}
-					submitText={submitText}
+		if (stripeKey && !isStripeInitialized) {
+			// stripe docs https://stripe.com/docs/stripe-js/react#elements-provider
+			loadStripe(stripeKey).then((newStripePromise) => setStripeInstance(newStripePromise));
+			setIsStripeInitialized(true);
+		}
+	}, [stripeKey, isStripeInitialized]);
+
+	const handlePayPal = async () => {
+		const isLoggedIn = await Identity.isLoggedIn();
+
+		if (!isLoggedIn) {
+			window.location.href = `${loginURL}?redirect=${checkoutURL}`;
+		}
+
+		const { country, email } = userInfo;
+			if (orderNumber) {
+				// Stripe Intents is set up and the payment was initialized already with stripe
+				// ... check is logged In
+				const currentOrder = await Sales.getOrderDetails(orderNumber);
+				if (currentOrder?.items?.length) {
+					Sales.clearCart().then(() => {
+						const items = currentOrder.items.map((item) => {
+							const { sku, priceCode, quantity } = item;
+							return { sku, priceCode, quantity };
+						});
```

Comment:
Is this being done just to filter out the other properties that can be on `item`?